packages/babel-plugin/src/utils/state-manager.js

```diff
@@ -630,6 +621,141 @@ export default class StateManager {
+        }
+        return basePath;
+      });
+
+    // Load aliases from package.json imports field
+    try {
+      const packageJsonPath = path.join(projectDir, 'package.json');
+      if (fs.existsSync(packageJsonPath)) {
+        const rawConfig: mixed = JSON5.parse(
+          fs.readFileSync(packageJsonPath, 'utf8'),
+        );
+        if (!isPackageJSON(rawConfig)) {
+          throw new Error('Invalid package.json format');
+        }
+        const packageJson: PackageJSON = rawConfig as $FlowFixMe;
+
+        // Handle Node.js native imports
+        const imports = packageJson.imports;
+        if (imports && typeof imports === 'object') {
+          packageAliases = Object.fromEntries(
+            Object.entries(imports)
+              .filter(([key]) => key.startsWith('#'))
+              .map(([key, value]) => [key.slice(1), resolveAliasPaths(value)]),
+          );
+        }
+      }
+    } catch (err) {
+      console.warn('Failed to load aliases from package.json:', err.message);
+    }
+
+    // Load aliases from tsconfig.json
+    try {
+      const tsconfigPath = path.join(projectDir, 'tsconfig.json');
+      if (fs.existsSync(tsconfigPath)) {
+        const rawConfig: mixed = JSON5.parse(
+          fs.readFileSync(tsconfigPath, 'utf8'),
+        );
+        if (!isTSConfig(rawConfig)) {
+          throw new Error('Invalid tsconfig.json format');
+        }
+        const tsconfig: TSConfig = rawConfig as $FlowFixMe;
+        const tsConfigAliasPaths = tsconfig.compilerOptions?.paths;
+        if (tsConfigAliasPaths != null && isImportsObject(tsConfigAliasPaths)) {
+          tsconfigAliases = Object.fromEntries(
+            Object.entries(tsConfigAliasPaths).map(([key, value]) => [
+              key,
+              resolveAliasPaths(value),
+            ]),
+          );
+        }
+      }
+    } catch (err) {
+      console.warn('Failed to load aliases from tsconfig.json:', err.message);
+    }
+
+    // Load aliases from deno.json
+    try {
+      const denoConfigPath = path.join(projectDir, 'deno.json');
+      if (fs.existsSync(denoConfigPath)) {
+        const rawConfig: mixed = JSON5.parse(
+          fs.readFileSync(denoConfigPath, 'utf8'),
+        );
+        if (!isDenoConfig(rawConfig)) {
+          throw new Error('Invalid deno.json format');
+        }
+        const denoConfig: DenoConfig = rawConfig as $FlowFixMe;
+        if (denoConfig.imports) {
+          denoAliases = Object.fromEntries(
+            Object.entries(denoConfig.imports).map(([key, value]) => {
+              return [key, resolveAliasPaths(value)];
+            }),
+          );
+        }
+      }
+    } catch (err) {
+      console.warn('Failed to load aliases from deno.json:', err.message);
+    }
+
+    // Merge aliases in priority: manual > package.json > tsconfig.json > deno.json
+    const mergedAliases = {
+      ...denoAliases,
+      ...tsconfigAliases,
+      ...packageAliases,
+      ...(manualAliases || {}),
```

Comment:
Do we still need manualAliases?

Reply:
I didn't want to remove it in the same commit, but we could probably remove them in the future once we are sure this feature is working for the wide variety of project types.

We could deprecate it in the next release and delete it entirely after.